### ISSUE TITLE
can't have different packages with same name for different debian/ubuntu releases

### ISSUE THREAD

**Issue by ajc** [rcrowley/freight#2]
I'm using FPM to build python packages, and while Ubuntu 10.04 and 10.10 have Python 2.6, 11.04 has 2.7 which means i need a separate package built for that.

If I try and use freight to serve up a single repository for all 3 Ubuntu releases, it seems that only one package "wins" (the 10.04 one, for what it's worth). When I try fetching the package on an 11.04 machine it (rightfully) barfs on a size mismatch.

**Comment by rcrowley** [comment]
I've run into this issue as well and as a stopgap, use version numbers like 1.2.3-1py2.7.  I guess it's time to split the `pool/` directory into something like `pool/lucid/main/...`.

**Comment by tmm1** [comment]
:+1:

**Comment by rcrowley** [comment]
Does this solve your problem, @ajc?  I no longer have to use the version number to publish Python 2.6- and 2.7-dependent packages.

**Comment by ajc** [comment]
I'll give it a whirl sometime today and let you know. Cheers!

**Comment by ajc** [comment]
Looks good!

I don't remember noticing this at the top of the output though:

```
% sudo freight cache apt/lucid apt/maverick apt/natty
[: 70: apt/lucid: unexpected operator
# [freight] pool already has buildbot-slave_0.8.3p1-1_all.deb
# [freight] pool already has buildbot_0.8.3p1-1_all.deb
```

...it only happens if i specify more than one distribution at once. ah, ok. if i replace the

```
[ -z "$@" ]
```

with

```
[ $# = 0 ]
```

it stops complaining.

**Comment by rcrowley** [comment]
Released in 0.0.8.
